presto-main/src/main/java/io/prestosql/operator/ExchangeClient.java

```diff
@@ -126,16 +130,32 @@
 
     // ExchangeClientStatus.mergeWith assumes all clients have the same bufferCapacity.
     // Please change that method accordingly when this assumption becomes not true.
-    public ExchangeClient(
-            DataSize bufferCapacity,
-            DataSize maxResponseSize,
-            int concurrentRequestMultiplier,
-            Duration maxErrorDuration,
-            boolean acknowledgePages,
-            HttpClient httpClient,
-            ScheduledExecutorService scheduler,
-            LocalMemoryContext systemMemoryContext,
-            Executor pageBufferClientCallbackExecutor)
+    public ExchangeClient(DataSize bufferCapacity,
+                          DataSize maxResponseSize,
+                          int concurrentRequestMultiplier,
+                          Duration maxErrorDuration,
+                          boolean acknowledgePages,
+                          HttpClient httpClient,
+                          ScheduledExecutorService scheduler,
+                          LocalMemoryContext systemMemoryContext,
+                          Executor pageBufferClientCallbackExecutor,
+                          FailureDetector failureDetector)
+    {
+        this(bufferCapacity, maxResponseSize, concurrentRequestMultiplier, maxErrorDuration, acknowledgePages, httpClient, scheduler, systemMemoryContext, pageBufferClientCallbackExecutor, failureDetector, true, 10);
```

Comment:
better we take maxRetryCount from config, instead of hardcoding

Reply:
fixed